cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -173,48 +183,35 @@ func (csr *ClusterStateRegistry) updateScaleRequests(currentTime time.Time) {
 	// clean up stale backoff info
 	csr.nodeGroupBackoffInfo.RemoveStaleBackoffData(currentTime)
 
-	timedOutSur := make([]*ScaleUpRequest, 0)
-	newSur := make([]*ScaleUpRequest, 0)
-	for _, sur := range csr.scaleUpRequests {
-		if !csr.areThereUpcomingNodesInNodeGroup(sur.NodeGroupName) {
+	for nodeGroupName, scaleUpRequest := range csr.scaleUpRequests {
+		if !csr.areThereUpcomingNodesInNodeGroup(nodeGroupName) {
 			// scale-out finished successfully
 			// remove it and reset node group backoff
-			csr.nodeGroupBackoffInfo.RemoveBackoff(sur.NodeGroupName)
+			csr.nodeGroupBackoffInfo.RemoveBackoff(nodeGroupName)
 			glog.V(4).Infof("Scale up in group %v finished successfully in %v",
-				sur.NodeGroupName, currentTime.Sub(sur.Time))
+				nodeGroupName, currentTime.Sub(scaleUpRequest.Time))
```

Comment:
Do we want to do `delete(csr.scaleUpRequests, nodeGroupName)` here as well, or is it cleaned up elsewhere?